src/cli/domain/get-components-by-dir.js

```diff
@@ -4,47 +4,41 @@ var fs = require('fs-extra');
 var path = require('path');
 var _ = require('underscore');
 
-module.exports = function(dependencies){
-  var logger = dependencies.logger;
-  return function(componentsDir, callback){
-
-    try {
-      var components = fs.readdirSync(componentsDir).filter(function(file){
-
-        var filePath = path.resolve(componentsDir, file),
-            isDir = fs.lstatSync(filePath).isDirectory(),
-            packagePath = path.join(filePath, 'package.json');
+module.exports = function(){
 
-        if(!isDir || !fs.existsSync(packagePath)){
-          return false;
-        }
+  return function(componentsDir, callback){
 
-        var content;
+    var isOcComponent = function(file){
 
-        try {
-          content = fs.readJsonSync(packagePath);
-        }
-        catch(err)
-        {
-          logger.log(('error reading ' + packagePath + ' ' + err.toString()).red);
-          return false;
-        }
+      var filePath = path.resolve(componentsDir, file),
+          packagePath = path.join(filePath, 'package.json'),
+          content;
 
-        if(!content.oc || !!content.oc.packaged){
-          return false;
-        }
+      try {
+        content = fs.readJsonSync(packagePath);
+      }
+      catch(err)
+      {
+        return false;
+      }
 
-        return true;
-      });
+      var packagedProperty = content.oc && content.oc.packaged;
 
-      var fullPathComponents = _.map(components, function(component){
-        return path.resolve(componentsDir, component);
-      });
+      return _.isUndefined(packagedProperty);
+    };
 
-      callback(null, fullPathComponents);
+    var dirContent;
 
+    try {
+      dirContent = fs.readdirSync(componentsDir);
     } catch(err){
-      return callback(err);
+      return callback(null, []);
```

Comment:
if the dir can't be read because is protected or doesn't exist, let's just say we didn't find anything.